Backported this patch from truffleruby 24.2.0 to 24.1.0 for compat with OpenSSL 3.4.x

diff --git a/truffleruby/lib/truffle/rbconfig.rb b/truffleruby/lib/truffle/rbconfig.rb
index 0b3b423712..6bf7518e09 100644
--- a/truffleruby/lib/truffle/rbconfig.rb
+++ b/truffleruby/lib/truffle/rbconfig.rb
@@ -144,6 +144,9 @@ module RbConfig

     warnflags << '-Wundef' # Warn for undefined preprocessor macros for core C extensions
     warnflags << '-Werror' # Make sure there are no warnings in core C extensions
+    # If there are deprecations in core C extensions, do not error for them.
+    # This would be problematic for extconf.rb checks as they would think such deprecated functions do not exist.
+    warnflags << '-Wno-error=deprecated-declarations'
   else
     libtruffleruby = "#{cext_dir}/libtruffleruby.#{soext}"
     libtrufflerubytrampoline = "#{cext_dir}/libtrufflerubytrampoline.#{soext}"
diff --git a/truffleruby/src/main/c/openssl/extconf.rb b/truffleruby/src/main/c/openssl/extconf.rb
index 76c4676331..694d590088 100644
--- a/truffleruby/src/main/c/openssl/extconf.rb
+++ b/truffleruby/src/main/c/openssl/extconf.rb
@@ -13,11 +13,6 @@

 require "mkmf"

-if defined?(::TruffleRuby)
-  # Needed with libssl 3.0.0 and -Werror from building core C extensions
-  $warnflags += ' -Wno-deprecated-declarations'
-end
-
 dir_config_given = dir_config("openssl").any?

 dir_config("kerberos")
